cdap-cli/src/main/java/co/cask/cdap/cli/command/CreateDatasetInstanceCommandV2.java

```diff
@@ -0,0 +1,92 @@
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.cli.command;
+
+import co.cask.cdap.cli.ArgumentName;
+import co.cask.cdap.cli.CLIConfig;
+import co.cask.cdap.cli.ElementType;
+import co.cask.cdap.cli.english.Article;
+import co.cask.cdap.cli.english.Fragment;
+import co.cask.cdap.cli.util.AbstractAuthCommand;
+import co.cask.cdap.cli.util.ArgumentParser;
+import co.cask.cdap.client.DatasetClient;
+import co.cask.cdap.proto.DatasetInstanceConfiguration;
+import co.cask.common.cli.Arguments;
+import com.google.gson.Gson;
+import com.google.inject.Inject;
+
+import java.io.PrintStream;
+import java.util.Map;
+
+/**
+ * Creates a dataset.
+ */
+public class CreateDatasetInstanceCommandV2 extends AbstractAuthCommand {
+
+  private static final Gson GSON = new Gson();
+  private final DatasetClient datasetClient;
+
+  @Inject
+  public CreateDatasetInstanceCommandV2(DatasetClient datasetClient, CLIConfig cliConfig) {
+    super(cliConfig);
+    this.datasetClient = datasetClient;
+  }
+
+  @Override
+  public void perform(Arguments arguments, PrintStream output) throws Exception {
+    String datasetType = arguments.get(ArgumentName.DATASET_TYPE.toString());
+    String datasetName = arguments.get(ArgumentName.NEW_DATASET.toString());
+    String datasetPropertiesString = arguments.getOptional(ArgumentName.DATASET_PROPERTIES.toString(), "");
+    String datasetDescription = arguments.getOptional(ArgumentName.DATASET_DESCRIPTON.toString(), null);
+    String datasetOwner = arguments.getOptional(ArgumentName.PRINCIPAL.toString(), null);
+
+    Map<String, String> datasetProperties = ArgumentParser.parseMap(datasetPropertiesString);
+    DatasetInstanceConfiguration datasetConfig =
+      new DatasetInstanceConfiguration(datasetType, datasetProperties, datasetDescription, datasetOwner);
+
+    datasetClient.create(cliConfig.getCurrentNamespace().dataset(datasetName), datasetConfig);
+    StringBuilder builder = new StringBuilder(String.format("Successfully created dataset named '%s' with type " +
+                                                              "'%s', properties '%s'", datasetName, datasetType,
+                                                            GSON.toJson(datasetProperties)));
+    if (datasetDescription != null) {
+      builder.append(String.format(", description '%s'", datasetDescription));
+    }
+    if (datasetOwner != null) {
+      builder.append(String.format(", owner principal '%s'", datasetOwner));
+    }
+    output.printf(builder.toString());
+    output.println();
+  }
+
+  @Override
+  public String getPattern() {
+    return String.format("v2 create dataset instance <%s> of type <%s> [with properties <%s>] [%s <%s>] " +
```

Comment:
Note: This v2 thing is temporary. We haven't yet come to a consensus as what the new command should start with in mandatory part. It cannot be same as ``create dataset instance`` because it same for the deprecated command and the command matcher doesn't work . 

See: https://issues.cask.co/browse/CDAP-8388